tutorials/yolov8_world_e2e.py

```diff
@@ -0,0 +1,405 @@
+    session = ort.InferenceSession(str(onnx_model_path), providers=providers, sess_options=session_options)
+
+    input_name = [i.name for i in session.get_inputs()]
+    if model_input == "image":
+        image_bytes = np.frombuffer(open(test_image, "rb").read(), dtype=np.uint8)
+        model_input = {input_name[0]: image_bytes}
+    else:
+        rgb_image = np.array(Image.open(test_image).convert("RGB"))
+        if rgb_layout == "CHW":
+            rgb_image = rgb_image.transpose((2, 0, 1))  # Channels first
+        model_input = {input_name[0]: rgb_image}
+
+    model_output = ["image"] if model_outputs_image else ["nms_output_with_scaled_boxes_and_keypoints"]
+    outputs = session.run(model_output, model_input)
+
+    if model_outputs_image:
+        # jpg or png with bounding boxes draw
+        image_out = outputs[0]
+        from io import BytesIO
+
+        s = BytesIO(image_out)
+        Image.open(s).show()
+    else:
+        # open original image so we can draw on it
+        input_image = Image.open(test_image).convert("RGB")
+        input_image_draw = ImageDraw.Draw(input_image)
+
+        scaled_nms_output = outputs[0]
+        for result in scaled_nms_output:
+            # split the 4 box coords, 1 score, 1 class
+            (box, score, class_id) = np.split(result, (4, 5))
+            class_id = int(class_id)
+            score = float(score * 100)
+
+            # convert box from centered XYWH to co-ords and draw rectangle
+            # NOTE: The pytorch model seems to output XYXY co-ords. Not sure why that's different.
+            half_w = box[2] / 2
+            half_h = box[3] / 2
+            x0 = box[0] - half_w
+            y0 = box[1] - half_h
+            x1 = box[0] + half_w
+            y1 = box[1] + half_h
+            input_image_draw.rectangle(((x0, y0), (x1, y1)), outline="red", width=4)
+            input_image_draw.text((x0, y0), f"{classes[class_id]}-{score:.2f}%")
+
+        print("Displaying original image with bounding boxes.")
+        input_image.show()
+
+
+# python yolov8_world_e2e.py yolov8m-worldv2.onnx --infer --test_image=image.jpg --input=rgb --input_shape H,W,3
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        """Add pre and post processing to the YOLOv8 World model. The model can be updated to take either 
+        jpg/png bytes as input (--input image), or RGB data (--input rgb).
+        NOTE: Use only YOLO WOrld v2 model as that is the only one with export capability.
+        By default the post processing will scale the bounding boxes and key points to the original image.
+    """
+    )
+    parser.add_argument("model", type=Path, help="The ONNX YOLOv8 World model.")
+    parser.add_argument(
+        "--classes",
+        type=Path,
+        default="classes.json",
+        help="JSON file containing list of classes that will be set as yolo world prompt.",
+    )
```

Comment:
Where should a user get this file from in order to run the script?

Reply:
In yolo world you specify prompts to the model. So, it's just a list of text/classes that you will prompt the model for zero-shot detection. So, it's user specified. The user has to create the list of prompts.

Reply:
Ideally a tutorial is self-contained with default inputs that allow running it end-to-end. We want it to make it as simple as possible for the user to see it working.

Is there an example classes.json that could be added that works with the default `--test_image` value of data/stormtroopers.jpg?

Reply:
I can add something. Should I create a file and commit it or have user pass in a list of values as classes and we can set default example classes for the argument?

Reply:
I have provided default values for classes as requested. Please review @skottmckay